tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+        });
+        it('send OPTIONS request with non-listed method "DELETE"', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'DELETE',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+    });
+    describe('CORS allows method POST and allows all origins', () => {
+        const corsParams = {
+            Bucket: bucket,
+            CORSConfiguration: {
+                CORSRules: [
+                    {
+                        AllowedMethods: [
+                            'POST',
+                        ],
+                        AllowedOrigins: [
+                            '*',
+                        ],
+                    },
+                ],
+            },
+        };
+        beforeEach(done => {
+            s3.putBucketCors(corsParams, done);
+        });
+
+        afterEach(done => {
+            s3.deleteBucketCors({ Bucket: bucket }, done);
+        });
+
+        it('send OPTIONS request with listed method "POST"', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'POST',
+            };
+            const headersResponse = {
+                'access-control-allow-origin': '*',
+                'access-control-allow-methods': 'POST',
+                'vary': 'Origin, Access-Control-Request-Headers, ' +
+                    'Access-Control-Request-Method',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 200,
+            headersResponse }, done);
+        });
+        it('send OPTIONS request with non-listed method: "PUT"', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'PUT',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+        it('send OPTIONS request with non-listed method: "DELETE"', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'DELETE',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+        it('send OPTIONS request with non-listed method "GET"', done => {
+            const headers = {
+                'Origin': 'http://www.allowedwebsite.com',
+                'Access-Control-Request-Method': 'GET',
+            };
+            methodRequest({ method: 'OPTIONS', headers, code: 'AccessForbidden',
+            headersResponse: null }, done);
+        });
+    });
+
+    describe('CORS allows method PUT and allows all origins', () => {
```

Comment:
if this is exactly the same as the describe above other than the method, it would probably be clearer to use a variable and combine the code (so a short forEach)